src/django_components/component.py

```diff
@@ -670,39 +669,37 @@ def _render_impl(
         cache_inlined_css(self.__class__, self.css or "")
 
         with _prepare_template(self, context, context_data) as template:
-            fills = self._escape_slot_fills(slots, escape_slots_content)
+            # For users, we expose boolean variables that they may check
+            # to see if given slot was filled, e.g.:
+            # `{% if variable > 8 and component_vars.is_filled.header %}`
+            is_filled = SlotIsFilled(slots_untyped)
+            self._render_stack[-1].is_filled = is_filled
 
-            _, resolved_fills = resolve_slots(
-                context,
-                template,
+            component_slot_ctx = ComponentSlotContext(
```

Comment:
How the slot rendering works now is that:

1. When we reach `Component.render()`, we set
    ```py
    context[_COMPONENT_SLOT_CTX_CONTEXT_KEY] = component_slot_ctx
    ```
   which holds data that slots need from the component (e.g. component name or available slot fills).
2. Then, as we're rendering the Component's content, we may come across `{% slot %}` tags (AKA `SlotNodes`).
3. Inside `SlotNode.render()`, we check if the `_COMPONENT_SLOT_CTX_CONTEXT_KEY` is accessible.
4. If it is, and there is a fill with the same name as the slot name, we render that fill in place of the slot.
5. If no fill matches, we render the slot's default content.